src/nvim/normal.c

```diff
@@ -3517,9 +3517,11 @@ static void nv_ident(cmdarg_T *cap)
       p = vim_strsave_shellescape(ptr, true, true);
     }
     xfree(ptr);
-    char *newbuf = xrealloc(buf, strlen(buf) + strlen(p) + 1);
+    const size_t buf_len = strlen(buf);
+    const size_t p_len = strlen(p);
+    char *newbuf = xrealloc(buf, buf_len + p_len + 1);
     buf = newbuf;
-    STRCAT(buf, p);
+    (void)STRCPY_LEN(buf + buf_len, p, p_len);
```

Comment:
This looks like a decrease in readability and also doesn't look like it gained any micro-optimization. What did we gain here?